game/src/plugins/skills/mining/src/mining.plugin.kts

```diff
@@ -0,0 +1,130 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.game.model.Position
+import org.apollo.game.model.entity.EquipmentConstants
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.Skill
+import org.apollo.game.plugin.skills.mining.*
+import kotlin.properties.Delegates
+
+class MiningAction : DistancedAction<Player> {
+
+    var ore: Ore by Delegates.notNull()
+    var counter: Int = 0
+    var started: Boolean = false
+
+    constructor(player: Player, position: Position, ore: Ore) : super(0, true, player, position, 1 /* ORE SIZE */) {
+        this.ore = ore
+    }
+
+    override fun executeAction() {
+        val level = mob.skillSet.getSkill(Skill.MINING).currentLevel
+        val pickaxe = findPickaxe()
+        mob.turnTo(position)
+
+        //check that our pick can mine the ore
+        if (pickaxe == null || level < pickaxe.level) {
+            mob.sendMessage("You do not have a pickaxe for which you have the level to use.")
+            stop()
+            return
+        }
+
+        //check that we can mine the ore
+        if (ore.level > level) {
+            mob.sendMessage("You do not have the required level to mine this rock.")
+            stop()
+            return
+        }
+
+        //start the process of mining
+        if (started) {
+            if (counter == 0) {
+                //TODO: calculate the chance that the player can actually get the ore
```

Comment:
I think this needs to be done before we can merge this.

Reply:
Does anyone know the chance that the player gets the ore? It was not in the ruby version of this plugin.

Reply:
Some info on that can be found here: http://runescape.wikia.com/wiki/Talk:Mining#Mining_success_rate_formula

Reply:
OK, implemented! Thank you, I was looking all over for this!